### PR TITLE
Feature/add custom geo areas

### PR DESCRIPTION
**Pull request by subinasr**
- Addresses #1783 

## Changes

* Add Custom geo area from Geo Area tab in new project flow

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

## This PR contains valid:

- [ ] permission checks
- [ ] translations

### CODE REVIEW

File: src/newViews/ProjectEdit/GeoAreas/CustomGeoAddForm/index.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+                    error={error?.title}
+                    label="Custom geo area title"
+                />
+                <TextInput
+                    className={styles.input}
+                    name="code"
+                    value={value.code}
+                    onChange={setFieldValue}
+                    error={error?.code}
+                    label="Custom geo area code"
```
Comment (tnagorra) [inline src/newViews/ProjectEdit/GeoAreas/CustomGeoAddForm/index.tsx]:
```suggestion
                    label="Title"
                />
                <TextInput
                    className={styles.input}
                    name="code"
                    value={value.code}
                    onChange={setFieldValue}
                    error={error?.code}
                    label="Code"
```

File: src/newViews/ProjectEdit/GeoAreas/RegionMapList/index.tsx [hunk 2]
```diff
@@ -1,37 +1,100 @@
...
+    const [regionOptions, setRegionOptions] = useState<Region[] | null | undefined>(undefined);
+    const [activeTab, setActiveTab] = useState<string>('');
+
+    const {
+        response: regionResponse,
+        retrigger: regionsGetTrigger,
+    } = useRequest<{ regions: Region[] }>({
+        url: `server://projects/${activeProject}/regions/`,
+        method: 'GET',
+        onSuccess: response => setActiveTab(response.regions[0]?.id.toString()),
```
Comment (tnagorra) [inline src/newViews/ProjectEdit/GeoAreas/RegionMapList/index.tsx]:
```suggestion
        onSuccess: (response) => {
            setActiveTab(response.regions[0]?.id.toString());
        },
```

File: src/newViews/ProjectEdit/GeoAreas/RegionMapList/index.tsx [hunk 3]
```diff
@@ -42,17 +105,20 @@ function RegionMapList(props: Props) {
...
-        );
-    }
-
     return (
         <div className={_cs(styles.regionMapList, className)}>
+            <RegionSelectInput
+                className={_cs(styles.input, styles.region)}
+                name="regions"
+                activeProject={activeProject}
+                value={selectedRegion}
```
Comment (tnagorra) [inline src/newViews/ProjectEdit/GeoAreas/RegionMapList/index.tsx]: I don't know if we need to pass value here

Comment (tnagorra) [reply src/newViews/ProjectEdit/GeoAreas/RegionMapList/index.tsx]: @AdityaKhatri what do you think?

File: src/newViews/ProjectEdit/GeoAreas/RegionMapList/index.tsx [hunk 4]
```diff
@@ -65,23 +131,30 @@ function RegionMapList(props: Props) {
...
-                            data={regions}
+                            data={regionResponse?.regions}
                             rendererParams={tabRendererParams}
                             renderer={Tab}
                             keySelector={regionKeySelector}
                         />
                     </TabList>
                 </Header>
                 <div className={styles.tabPanelContainer}>
+                    {isDefined(regionResponse) && regionResponse?.regions.length < 1 && (
```
Comment (tnagorra) [inline src/newViews/ProjectEdit/GeoAreas/RegionMapList/index.tsx]:
```suggestion
                    {regionResponse && regionResponse?.regions.length < 1 && (
```

**Review by tnagorra — CHANGES_REQUESTED** [review]

**Review by samshara — APPROVED** [review]

**Comment by samshara** [comment]
@subinasr https://github.com/the-deep/client/pull/1944 this pull request seems to have the same commit. If so please create a single pull request.

File: src/newViews/ProjectEdit/GeoAreas/CustomGeoAddForm/index.tsx (hunk 1 above)
Comment (subinasr) [reply src/newViews/ProjectEdit/GeoAreas/CustomGeoAddForm/index.tsx]: That was as per the mockup.
image

File: src/newViews/ProjectEdit/GeoAreas/RegionMapList/index.tsx (hunk 3 above)
Comment (subinasr) [reply src/newViews/ProjectEdit/GeoAreas/RegionMapList/index.tsx]: I think you're right, we don't really need the value.

**Pull request merged**